catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/AbstractAssociatedMetacardsHandler.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.workspace.transformer.impl;
+
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.impl.CreateRequestImpl;
+import ddf.catalog.operation.impl.DeleteRequestImpl;
+import ddf.catalog.operation.impl.UpdateRequestImpl;
+import ddf.catalog.source.IngestException;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractAssociatedMetacardsHandler {
+
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(AbstractAssociatedMetacardsHandler.class);
+
+  protected CatalogFramework catalogFramework;
+
+  protected EndpointUtil util;
+
+  protected FilterBuilder filterBuilder;
+
+  public AbstractAssociatedMetacardsHandler(
+      CatalogFramework catalogFramework, EndpointUtil util, FilterBuilder filterBuilder) {
+    this.catalogFramework = catalogFramework;
+    this.util = util;
+    this.filterBuilder = filterBuilder;
+  }
+
+  public void handle(Metacard existing, Metacard updated, List<Metacard> queries) {
+    create(existing, updated, queries);
```

Comment:
Could lead to unintended behavior with creating, deleting, and then trying to update some query metacard.